### ISSUE TITLE
Removing all cached data

### ISSUE THREAD

**Issue by hamzaozturk** [Haneke/Haneke#71]
Is there any way to remove all cached data for each app launching?

**Comment by jmartos89** [comment]
in my case the following code works perfectly:
 [[HNKCache sharedCache] removeAllImages];

**Comment by bdorfman** [comment]
removeAllImages only removes caches for registered formats. If you want to make sure everything is completely dropped, you'd need to manually delete the entire cache directory. removeAllImages should probably be changed to do this, as the name is currently misleading.

**Comment by kartikthapar** [comment]
`removeAllImages` doesn't work for me when using disk cache. It doesn't seem to delete any images.

**Comment by mattmorton** [comment]
My understanding is that `removeAllImages` is only available to HNKCache, not HNKDiskCache?

**Comment by mattmorton** [comment]
I've been back and checked this, and my understanding was incorrect.

**Comment by bdorfman** [comment]
@kartikthapar The implementation of removeAlImages in this repo only removes formats which have been registered prior to you calling that method. Stick a breakpoint in and see if this is your problem (eg format list is empty).

**Comment by emergencyhug** [comment]
Sometimes there would be some images remain after [[HNKCache sharedCache] removeAllImages]. I tried to remove all files in Cache folder by using the NSFileManager but no more images could be write into Haneke cache due to the lack of folder directory.
I think Haneke shoud be checking and recreating folders if they're not exist?
Error logged out in console (test on simulator) : Failed to write to file Error Domain=NSCocoaErrorDomain Code=4 "The operation couldn’t be completed. (Cocoa error 4.)" UserInfo=0x7f860bae7ab0 {NSFilePath= /.../...image file path "The operation couldn’t be completed. No such file or directory"})

**Comment by bdorfman** [comment]
Yes, you have to recreate the folder if it doesn't exist. The Foursquare fork of Haneke adds all of this functionality if you would just like to use our version.

**Comment by emergencyhug** [comment]
@bdorfman thanks and your version works fine!  Just curious, how did you guys recreate the folders holding different formatted images?

**Comment by bdorfman** [comment]
Whenever our HNKDiskCache tries to write an image to disk and it fails because the directory doesn't exist, it just creates the directory and tries again. Seems to work fine, and fixes some other issues we were having where the directories disappeared (I think because the OS automatically cleaned up our caches directory because of full disk space).
